src/Compilers/CSharp/Portable/BoundTree/DecisionTreeBuilder.cs

```diff
@@ -108,14 +105,11 @@ private DecisionTree AddByValue(DecisionTree.Guarded guarded, BoundConstantPatte
         {
             if (guarded.Default != null)
             {
-                if (guarded.Default.MatchIsComplete)
-                {
-                    return null;
-                }
+                Debug.Assert(!guarded.Default.MatchIsComplete); // otherwise we would have given a subsumption error
```

Comment:
I don't think that is necessary. The called code following should handle the situation. The purpose of this `if` is to create a decision tree for `guarded.Default` if it has none.

---
In reply to: [100589692](https://github.com/dotnet/roslyn/pull/16928#discussion_r100589692) [](ancestors = 100589692)